native/dgels.go

```diff
@@ -0,0 +1,204 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import (
+	"github.com/gonum/blas"
+	"github.com/gonum/lapack"
+)
+
+// Dgels finds a minimum-norm solution based on the matrices a and b using the
+// QR or LQ factorization to find the solution. Dgels returns false if the matrix
+// A is singular, and true if this solution was successfully found.
```

Comment:
I don't think so (barring floating point issues). Dormqr cannot fail. Only the triangular solve can fail, and that only fails if the diagonal entry is zero. If the matrix is really badly conditioned you could have some Infs from floating point issues, but ok would still be true because it is not detected in this function.